test/powershell/Modules/Microsoft.PowerShell.Management/Clear-Content.Tests.ps1

```diff
@@ -80,40 +80,52 @@ Describe "Clear-Content cmdlet tests" -Tags "CI" {
     }
 
-    It "Alternate streams should be cleared with clear-content" -Skip:(!$IsWindows) {
-      # make sure that the content is correct
-      # this is here rather than BeforeAll because only windows can write to an alternate stream
-      Set-Content -Path "TestDrive:/$file3" -Stream $streamName -Value $streamContent
-      Get-Content -Path "TestDrive:/$file3" | Should -BeExactly $content2
-      Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeExactly $streamContent
-      Clear-Content -Path "TestDrive:/$file3" -Stream $streamName
-      Get-Content -Path "TestDrive:/$file3" | Should -BeExactly $content2
-      Get-Content -Path "TestDrive:/$file3" -Stream $streamName | Should -BeNullOrEmpty
-    }
-    It "Should not error when targeting an existing stream on a directory." -Skip:(!$IsWindows) {
-      { Set-Content -Path "TestDrive:/$dirName" -Stream $streamName -Value $streamContent } | Should -Not -Throw
-      Get-Content -Path "TestDrive:/$dirName" -Stream $streamName | Should -BeExactly $streamContent
-      { Clear-Content -Path TestDrive:\$dirName -Stream $streamName -ErrorAction Stop } | Should -Not -Throw
-    }
-
-    It "the '-Stream' dynamic parameter is visible to get-command in the filesystem" -Skip:(!$IsWindows) {
-      try {
-        Push-Location -Path TestDrive:
-        (Get-Command Clear-Content -Stream foo).parameters.keys -eq "stream" | Should -Be "stream"
+    Context "Clear-Content should work with alternate data streams on Windows" {
+      BeforeAll {
+        if (!$IsWindows)
+        {
+          return
+        }
+        Set-Content -Path "TestDrive:/$file3" -Stream $streamName -Value $streamContent
+        Set-Content -Path "TestDrive:/$dirName" -Stream $streamName -Value $streamContent
+      }
+      It "Alternate streams should be cleared with Clear-Content" -Skip:(!$IsWindows) {
+        # Make sure the test is set up correctly.
+        Set-Content           -Path "TestDrive:/$file3" -Stream $streamName -Value $streamContent
+        # Truncate the stream.
+        Clear-Content         -Path "TestDrive:/$file3" -Stream $streamName -ErrorAction Stop
+        # The stream should exist, but should have 0 length.
+        $result = Get-Item -Path "TestDrive:/$file3" -Stream $streamName
+        $result | Should -Not -BeNullOrEmpty
+        $result.length | Should -Be 0
+      }
+      It "Should not error when clearing an alternate stream on a directory" -Skip:(!$IsWindows) {
+        # Make sure the test is set up correctly.
+        Set-Content           -Path "TestDrive:/$dirName" -Stream $streamName -Value $streamContent
+        # Truncate the stream.
+        Clear-Content         -Path "TestDrive:/$dirName" -Stream $streamName -ErrorAction Stop
+        # The stream should exist, but should have 0 length.
+        $result = Get-Item -Path "TestDrive:/$dirName" -Stream $streamName
+        $result | Should -Not -BeNullOrEmpty
+        $result.length | Should -Be 0
```

Comment:
I wonder - it is not empty but length is 0. What is a type of the `result` variable?

Reply:
Maybe make the check explicit? I mean the test looks like a trick. If we check that the return type is an array with 0 length this will more easy understand for reviewer

Reply:
Replaced with an explicit `| Should -BeOfType System.Management.Automation.Internal.AlternateStreamData`.